package.json

```diff
@@ -77,16 +77,25 @@
   },
   "devDependencies": {
     "@babel/core": "7.22.8",
+    "@babel/eslint-parser": "7.22.15",
+    "@babel/preset-env": "7.22.7",
     "@babel/runtime": "7.22.6",
-    "babel-eslint": "10.1.0",
-    "babel-jest": "26.6.3",
-    "eslint": "7.32.0",
+    "@jest/globals": "29.7.0",
+    "@react-native/eslint-config": "0.72.2",
+    "@react-native/metro-config": "0.72.11",
+    "@tsconfig/react-native": "3.0.2",
+    "@types/jest": "29.5.5",
+    "@types/react": "18.0.24",
+    "@types/react-test-renderer": "18.0.1",
+    "babel-jest": "29.7.0",
+    "eslint": "8.51.0",
     "eslint-config-airbnb": "19.0.4",
     "eslint-plugin-import": "2.27.5",
     "eslint-plugin-jsx-a11y": "6.7.1",
     "eslint-plugin-react": "7.32.2",
-    "jest": "26.6.3",
-    "metro-react-native-babel-preset": "0.72.4",
+    "jest": "29.7.0",
+    "metro-react-native-babel-preset": "0.76.8",
+    "prettier": "2.8.8",
```

Comment:
This is not on the acceptance criteria, why are we adding it here? Is it a requirement for react-native 0.72?

Reply:
These dependencies were suggested by the [React Native community `upgrade-helper`](https://react-native-community.github.io/upgrade-helper/?from=0.70.9&to=0.72.4#RnDiffApp-package.json).
- `jest` was actually updated by #332 but, due to a technicality, had to be temporarily reverted until this PR
- `metro-react-native-babel-preset` is needed to properly configure babel to the new `metro` version
- `prettier` is the one that's not actually _needed_, but I believe it's good to have it for a few reasons:
  - Improved code quality and common standards
  - Improved maintainability for future React Native upgrades: following the upgrade helper guide is quite helpful to have the most up-to-date mutually compatible dependencies

Reply:
I don't think we should add it in this PR as it is a new dependency and not related to the react-native upgrade, and I also don't think we should add new dependencies before LavaMoat (or similar) is implemented on this project